.gitignore

```diff
@@ -0,0 +1,17 @@
+# Mac stuff
+.DS_Store
+
+# Compiled Object files
+*.slo
+*.lo
+*.o
+
+# Compiled Dynamic libraries
+*.so
+*.dylib
+
+# Compiled Static libraries
+*.lai
+*.la
+*.a
+
```

Comment:
This is to prevent anyone from committing compiled files that would otherwise cause merge issues, I' m not sure which files are created during Arduino compilations but this file covers the normal C/C++ ones.

DS_Store is Mac OS X specific (folder metadata) but it won't harm anything, on the contrary it might be worth adding more stuff for different OS:es.